test.e2e/spec/services.spec.js

```diff
@@ -181,6 +181,89 @@ define(['angular', 'given', 'util'], function(angular, given, util) {
             }
           );
         });
+
+        it('intercepts internal requests with internal urlBase',function() {
+          var urlBase = '/api',
+              nonUrlBase = '/otherlocalpath',
+              authHeader = 'X-Test-Token',
+              accessTokenId = '12345';
+
+          loopBackResourceProvider.setUrlBase(urlBase);
+          loopBackResourceProvider.setAuthHeader(authHeader);
+
+          var $injector = createInjector();
+
+          // set custom accessTokenId
+          var auth = $injector.get('LoopBackAuth');
+          auth.accessTokenId = accessTokenId;
+
+          var $http = $injector.get('$http');
```

Comment:
L191-L200 are repeated in the test below too. Please extract them to a shared utility/helper method.

Example:

```js
function setupAuthenticatedHttpClient(urlBase, authHeader, tokenId) {
  loopBackResourceProvider.setUrlBase(urlBase);
  loopBackResourceProvider.setAuthHeader(authHeader);
  var $injector = createInjector();
  var auth = $injector.get('LoopBackAuth');
  auth.accessTokenId = accessTokenId;
  return $injector.get('$http');
}
```

Alternatively move (some of) these lines to a `beforeEach` hook. For example, I think it makes a lot of sense to call `setUrlBase` and `setAuthHeader` from a `beforeEach` hook.